CMakeLists.txt

```diff
@@ -1289,7 +1289,11 @@ set_property(TARGET   ${ZEPHYR_PREBUILT_EXECUTABLE} PROPERTY LINK_DEPENDS ${PROJ
 add_dependencies(     ${ZEPHYR_PREBUILT_EXECUTABLE} ${ALIGN_SIZING_DEP} ${PRIV_STACK_DEP} ${LINKER_SCRIPT_TARGET} ${OFFSETS_LIB})
 
 
-if(GKOF OR GKSF)
+if(NOT GKOF AND NOT GKSF)
```

Comment:
IMHO

`if(GKOF AND GKSF)
`
is faster to understand than

`if(NOT GKOF AND NOT GKSF)
`

perhaps we could instead create a second if-else block just for 

`set(logical_target_for_zephyr_elf`

to make it less hidden?

Reply:
I think the acronyms much more to blame than the boolean operators, at least for anyone new to this part of code ... how about rephrasing the comment on the line below like this:
```
# Use the prebuilt elf as the final elf because neither Kernel Objects
# nor Kernel Sources Files were Generated
```

Reply:
Here's yet another option:
```
set(EXTRA_LINKER_PASS  GKOF OR GKSF)

if(NOT EXTRA_LINKER_PASS)
...
````

Reply:
I'd prefer clean code over well documented code.

If it's the obscurity of GKOF AND GKSF that is causing readability issues then perhaps we could do:

```
if(GKOF OR GKSF)
set(use_zephyr_prebuilt_as_the_the_final_elf 0)
else()
set(use_zephyr_prebuilt_as_the_the_final_elf 1)
endif()
```

Reply:
`set(EXTRA_LINKER_PASS  GKOF OR GKSF)
`

This is not valid CMake unfortunately, an if-statement is needed.

Reply:
The problem with EXTRA_LINKER_PASS, is that we can have extra linker passes without EXTRA_LINKER_PASS being set, due to the use of 'construct_add_custom_command_for_linker_pass'.

Reply:
> This is not valid CMake unfortunately,..

This was pseudo-code shot from the hip. Here's a proper, successfully tested version:

```
set(generated_kernel_files ${GKSF} ${GKOF})
if(NOT generated_kernel_files)
  ...
else()
 ...
```
If you don't like this one and we can't quickly find common ground then let me know and I'll just drop this part of the change. The real, main issue wasn't there, it was the misleading libzephyr comment.

Reply:
This looks good to me.